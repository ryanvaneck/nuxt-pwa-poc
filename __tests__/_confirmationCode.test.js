import { mount } from '@vue/test-utils';
import ConfirmationCode from '../pages/_confirmationCode/index.vue';

const getOptions = (data) => ({
  data: function () {
    return data;
  }
});

const testData = {
  reservation: {
    reservation: {
      confirmationCode: 'AAAAAA',
      checkInDate: '2020-01-10',
      checkOutDate: '2020-02-14',
      city: 'San Francisco, CA',
      rating: null,
      cityImage:
        'https://cdn.pixabay.com/photo/2013/11/13/21/14/san-francisco-210230_960_720.jpg',
      locationDetail: 'Kasa Sand Diego at Aperture\nApartment 203'
    },
  },
  error: {
    error: 'No internet connection detected.'
  }
};

describe('ConfirmationCode', () => {
  it('should match snapshot on first render', async () => {
    const wrapper = mount(ConfirmationCode, getOptions(testData.reservation));
    await wrapper.vm.$nextTick();
    expect(wrapper.html()).toMatchSnapshot('first');
  });

  it('should match snapshot on no internet error', async () => {
    const wrapper = mount(ConfirmationCode, getOptions(testData.error));
    await wrapper.vm.$nextTick();
    expect(wrapper.html()).toMatchSnapshot('noInternetError');
  });
});

