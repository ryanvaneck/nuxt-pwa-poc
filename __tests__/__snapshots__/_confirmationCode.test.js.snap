// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmationCode should match snapshot on first render: first 1`] = `
"<div>
  <div>header</div>
  <div class=\\"reservation-page\\">
    <div>
      <div class=\\"reservation\\">
        <h1>Your reservation in San Francisco, CA</h1>
        <div class=\\"image\\"><img src=\\"https://cdn.pixabay.com/photo/2013/11/13/21/14/san-francisco-210230_960_720.jpg\\" alt=\\"San Francisco, CA\\" class=\\"city-image\\"></div>
        <div class=\\"details\\">
          <div class=\\"location\\">
            <h3>Location</h3>
            Kasa Sand Diego at Aperture
            Apartment 203
          </div>
          <div class=\\"dates\\">
            <h3>Dates</h3>
            Jan 9 2020 - Feb 13 2020 (35 days)
          </div>
          <div class=\\"confirmation-code\\">
            <h3>Confirmation Code</h3>
            #AAAAAA
          </div>
        </div>
      </div>
      <div class=\\"rating\\">
        <h3>Rate Your Experience!</h3>
        <div></div>
        😢 😔 😐 🙂 😊
      </div>
    </div>
    <!---->
  </div>
  <div class=\\"footer\\">footer</div>
</div>"
`;

exports[`ConfirmationCode should match snapshot on no internet error: noInternetError 1`] = `
"<div>
  <div>header</div>
  <div class=\\"reservation-page\\">
    <!---->
    <div> No internet connection detected.</div>
  </div>
  <div class=\\"footer\\">footer</div>
</div>"
`;
